src/keygen/participant.rs

```diff
@@ -126,11 +130,41 @@ pub struct KeygenParticipant {
     status: Status,
 }
 
+/// Output type from key generation, including all parties' public key shares,
+/// this party's private key share, and a bit of global randomness.
+#[derive(Debug, Clone)]
+pub struct Output {
+    public_key_shares: Vec<KeySharePublic>,
+    private_key_share: KeySharePrivate,
+    #[allow(unused)]
+    rid: [u8; 32],
+}
+
+impl Output {
+    /// Construct the generated public key.
+    pub fn public_key(&self) -> Result<VerifyingKey> {
+        let mut vk_point = CurvePoint::IDENTITY;
+        for keyshare in &self.public_key_shares {
+            vk_point = CurvePoint(vk_point.0 + keyshare.X.0);
```

Comment:
~Question about this loop iteration... It seems to just be assignment of `vk_point` to `CurvePoint(...)` is there a missing addition or something here? Otherwise the value of `vk_point` is just equal to the last element in the `public_key_shares`~

Reply:
Oh wait, I see that you are using `vk_point.0` in the iteration NVM!

Reply:
I mildly rephrased this to try to make it more obvious what's happening.